src/functionaltests/java/com/ericsson/ei/notifications/ttl/TestTTLSteps.java

```diff
@@ -42,196 +42,197 @@
+        assertEquals(2, requests.length());
+    }
+
+    @Then("^Check missed notification is in database$")
+    public void check_missed_notification_is_in_database() {
+        String condition = "{\"subscriptionName\" : \"" + SUBSCRIPTION_NAME + "\"}";
+        List<String> result = mongoDBHandler.find(missedNotificationDatabase, missedNotificationCollection, condition);
+
+        assertEquals(1, result.size());
+        assertEquals("Could not find a missed notification matching the condition: " + condition,
+                "\"" + SUBSCRIPTION_NAME + "\"", dbManager.getValueFromQuery(result, "subscriptionName", 0));
+    }
+
+    @Given("^A subscription is created using \"([^\"]*)\" with non-working notification meta$")
+    public void a_subscription_is_created_at_the_end_point_with_non_existent_notification_meta(String endPoint)
+            throws Throwable {
+        String readFileToString = FileUtils.readFileToString(new File(SUBSCRIPTION_FILE_PATH_CREATION), "UTF-8");
+        JSONArray jsonArr = new JSONArray(readFileToString);
+        httpRequest = new HttpRequest(HttpMethod.POST);
+        httpRequest.setHost(hostName).setPort(applicationPort).setEndpoint(endPoint)
+                .addHeader("content-type", "application/json").addHeader("Accept", "application/json")
+                .setBody(jsonArr.toString());
+        httpRequest.performRequest();
+    }
+
+    @Given("^I send an Eiffel event$")
+    public void eiffel_events_are_sent() throws Throwable {
+        LOGGER.debug("Sending an Eiffel event");
+        List<String> eventNamesToSend = getEventNamesToSend();
+        eventManager.sendEiffelEvents(EIFFEL_EVENTS_JSON_PATH, eventNamesToSend);
+        List<String> missingEventIds = dbManager
+                .verifyEventsInDB(eventManager.getEventsIdList(EIFFEL_EVENTS_JSON_PATH, eventNamesToSend));
+        assertEquals("The following events are missing in mongoDB: " + missingEventIds.toString(), 0,
+                missingEventIds.size());
+        LOGGER.debug("Eiffel event is sent");
+    }
+
+    @When("^Aggregated object is created$")
+    public void aggregated_object_is_created() throws Throwable {
+        // verify that aggregated object is created and present in db
+        LOGGER.debug("Checking presence of aggregated Object");
+        List<String> allObjects = mongoDBHandler.getAllDocuments(dataBase, collection);
+        assertEquals(1, allObjects.size());
+    }
+
+    @When("^Missed notification is created$")
+    public void a_missed_notification_is_created() throws Throwable {
+        // verifying that missed notification is created and present in db
+        String condition = "{\"subscriptionName\" : \"" + SUBSCRIPTION_NAME_3 + "\"}";
+        long maxTime = System.currentTimeMillis() + MAX_WAIT_TIME;
+        List<String> notificationExit = null;
+        LOGGER.debug("Checking presence of missnotification in db");
+        while (System.currentTimeMillis() < maxTime) {
```

Comment:
Please extract the piece of code that fetches existing notifications and create a method of it. Then reuse it in this and in the following step. 

Fix variable naming from notificationExit to notificationExist.

Reply:
Although this is out of the scope (if you disable white space changes in diff settings) I may look in to this.